### PR TITLE
LPS-196367 | Poshi Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-194499 - Product QA | Test Automation Creation

Hi! :)
This pr is corresponding to this ticket: [LPS-194499](https://liferay.atlassian.net/browse/LPS-194499) → [LPS-196367](https://liferay.atlassian.net/browse/LPS-196367)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase [hunk 1]
```diff
@@ -846,4 +846,178 @@ definition {
...
+		task ("And the link opens the content page in a new tab") {
+			SelectWindow.selectPopUp(value1 = "null");
+
+			ControlMenu.viewHeaderDraftTitle(headerTitle = "Test Page Name");
+		}
+	}
+
+	@description = "This is a test for LPS-182022. In the list of locked pages, the name of the Collection page, Master page, Content page template, Display page template are clickable and the link opens the pages in a new tab."
+	@priority = 5
+	test ViewPageNameClickableAndOpensInANewTab {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase:900]: NavigateToLockedPageTemplatesInList

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase [hunk 2]
```diff
@@ -846,4 +846,178 @@ definition {
...
+			ContentPages.addCollectionPage(
+				collectionName = "Dynamic Collection",
+				pageName = "Collection Page Name",
+				tabName = "Collections");
+
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				dropTarget = "Header",
+				fragmentName = "Button",
+				targetFragmentName = "Container");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase:942]: Move to NavigateToLockedPagesInList case

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 3]
```diff
@@ -146,6 +146,54 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-194499. Using keyboard navigation, the name of the Utility page in the Locked Pages list is clickable and opens the Utility page in a new tab."
+	@priority = 4
+	test NavigateToLockedPageInListViaKeyboard {
+		task ("Given a locked utility page") {
+			PagesAdmin.openPagesAdmin(siteURLKey = ${randomSiteName});
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:153]: "Test Site Name"

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 4]
```diff
@@ -146,6 +146,54 @@ definition {
...
+
+			Open.openInTheNewTab(value1 = "${baseURL}/web/${randomSiteName}");
+
+			ProductMenu.gotoPortlet(
+				category = "Configuration",
+				portlet = "Locked Pages");
+
+			LexiconTable.viewTableEntryPartialText(
+				tableEntry = "404 Error Utility Page",
+				tableEntryPartialText = "Seconds Ago");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:175]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 5]
```diff
@@ -146,6 +146,54 @@ definition {
...
+		}
+
+		task ("Then can click on the name link of the blocked page just using the keyboard") {
+			KeyPress(
+				key_columnName = "name",
+				key_tableEntry = "404 Error Utility Page",
+				locator1 = "LexiconTable#TABLE_ENTRY_COLUMN_LINK",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:182]: Use the previous element

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 6]
```diff
@@ -146,6 +146,54 @@ definition {
...
+
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				fragmentName = "Button");
+		}
+
+		task ("When navigating to the locked pages admin page") {
+			var baseURL = PropsUtil.get("portal.url");
+
+			Open.openInTheNewTab(value1 = "${baseURL}/web/${randomSiteName}");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:167]: test-site-name

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 7]
```diff
@@ -80,6 +80,55 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-194499. In the list of locked pages, the name of the content page is clickable and the link opens the content page in a new tab."
+	@priority = 5
+	test ViewLockedPageNameClickableAndOpensDraftContentPage {
+		property custom.properties = "feature.flag.LPS-180328=true";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:86]: ff is removed

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 8]
```diff
@@ -80,6 +80,55 @@ definition {
...
+
+			ProductMenu.gotoPortlet(
+				category = "Configuration",
+				portlet = "Locked Pages");
+
+			LexiconTable.viewTableEntryTitle(tableEntry = "Test Page Name");
+
+			LexiconTable.viewTableEntryPartialText(
+				tableEntry = "Test Page Name",
+				tableEntryPartialText = "Seconds Ago");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:116]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase [hunk 9]
```diff
@@ -270,6 +270,112 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-182022. In the list of locked pages, the name of the Collection page, Master page, Content page template, Display page template are clickable and the link opens the pages in a new tab."
+	@priority = 5
+	test NavigateToLockedPageTemplatesInList {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase:275]: Move to LockedPages.testcase

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase [hunk 10]
```diff
@@ -270,6 +270,112 @@ definition {
...
+			ContentPages.addCollectionPage(
+				collectionName = "Recent Content",
+				pageName = "Collection Page Name",
+				tabName = "Collection Providers");
+
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				dropTarget = "Header",
+				fragmentName = "Button",
+				targetFragmentName = "Container");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase:288]: Move collection page to NavigateToLockedPagesInList

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 11]
```diff
@@ -80,6 +80,55 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-194499. In the list of locked pages, the name of the content page is clickable and the link opens the content page in a new tab."
+	@priority = 5
+	test ViewLockedPageNameClickableAndOpensDraftContentPage {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:85]: Rename to NavigateToLockedPagesInList

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 12]
```diff
@@ -146,6 +146,54 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-194499. Using keyboard navigation, the name of the Utility page in the Locked Pages list is clickable and opens the Utility page in a new tab."
+	@priority = 4
+	test NavigateToLockedPageInListViaKeyboard {
+		task ("Given a locked utility page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:152]: Insert 		var siteName = selenium.getText("Home#SITE_NAME");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 13]
```diff
@@ -146,6 +146,54 @@ definition {
...
+				locator1 = "LexiconTable#TABLE_ENTRY_COLUMN_LINK",
+				value1 = "\TAB");
+
+			KeyPress(
+				key_columnName = "name",
+				key_tableEntry = "404 Error Utility Page",
+				locator1 = "LexiconTable#TABLE_ENTRY_COLUMN_LINK",
+				value1 = "\ENTER");
+
+			SelectWindow(value1 = "title=404 Error Utility Page - ${randomSiteName} - Liferay DXP");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:191]: SelectWindow(value1 = "title=404 Error Utility Page - Test Site Name - ${siteName}");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 14]
```diff
@@ -80,6 +80,55 @@ definition {
...
+		}
+
+		task ("Then the name of the content page is clickable") {
+			LexiconTable.clickTableEntryColumnLink(
+				columnName = "name",
+				tableEntry = "Test Page Name");
+		}
+
+		task ("And the link opens the content page in a new tab") {
+			SelectWindow.selectPopUp(value1 = "null");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:126]: SelectWindow(value1 = "title=404 Error Utility Page - Test Site Name - ${siteName}");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 15]
```diff
@@ -80,6 +80,55 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-194499. In the list of locked pages, the name of the content page is clickable and the link opens the content page in a new tab."
+	@priority = 5
+	test ViewLockedPageNameClickableAndOpensDraftContentPage {
+		property custom.properties = "feature.flag.LPS-180328=true";
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:87]: Insert var siteName = selenium.getText("Home#SITE_NAME");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase [hunk 16]
```diff
@@ -270,6 +270,112 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-182022. In the list of locked pages, the name of the Collection page, Master page, Content page template, Display page template are clickable and the link opens the pages in a new tab."
+	@priority = 5
+	test NavigateToLockedPageTemplatesInList {
+		task ("Given locked pages of various types") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase:276]: Insert var siteName = selenium.getText("Home#SITE_NAME");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase [hunk 17]
```diff
@@ -270,6 +270,112 @@ definition {
...
+					tableEntry = ${pageName},
+					tableEntryPartialText = "Seconds Ago");
+			}
+
+			for (var pageName : list "Collection Page Name,Master Page Name,Page Template Name,DPT Name") {
+				LexiconTable.clickTableEntryColumnLink(
+					columnName = "name",
+					tableEntry = ${pageName});
+
+				SelectWindow(value1 = "title=${pageName} - Test Site Name - Liferay DXP");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase:370]: SelectWindow(value1 = "title=${pageName} - Test Site Name - ${siteName}");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase [hunk 18]
```diff
@@ -270,6 +270,112 @@ definition {
...
+			ProductMenu.gotoPortlet(
+				category = "Configuration",
+				portlet = "Locked Pages");
+		}
+
+		task ("Then in the list of locked pages names are clickable and the link opens the pages in a new tab") {
+			for (var pageName : list "Collection Page Name,Master Page Name,Page Template Name,DPT Name") {
+				LexiconTable.viewTableEntryPartialText(
+					tableEntry = ${pageName},
+					tableEntryPartialText = "Seconds Ago");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase:362]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase [hunk 19]
```diff
@@ -270,6 +270,112 @@ definition {
...
+			for (var pageName : list "Collection Page Name,Master Page Name,Page Template Name,DPT Name") {
+				LexiconTable.clickTableEntryColumnLink(
+					columnName = "name",
+					tableEntry = ${pageName});
+
+				SelectWindow(value1 = "title=${pageName} - Test Site Name - Liferay DXP");
+
+				ControlMenu.viewHeaderDraftTitle(headerTitle = ${pageName});
+
+				Navigator.closeWindow();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pageeditor/PageEditorWithLockedPages.testcase:374]: Close.closeAndSelectWindow(value1 = "title=Locked Pages - Test Site Name - ${siteName}");

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @rafaellajordao, would you mind updating according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase (hunk 5 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:182]: Hey Tim, When I tried use the previous element I receive this error message `org.openqa.selenium.ElementNotInteractableException: element not interactable`
`//tr[@data-qa-id='row'][contains(.,'404 Error Utility Page')]//td[contains(@class,'name-column')]`
Can you help me with this?

**Comment by Tim-Cao** [comment]
@rafaellajordao, see changes https://github.com/liferay-echo/liferay-portal/pull/14389

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**